RecoLocalCalo/HGCalRecAlgos/test/alpaka/TestHGCalRecHitESProducers.cc

```diff
@@ -0,0 +1,139 @@
+// Author: Izaak Neutelings (March 2024)
+// Based on: https://github.com/CMS-HGCAL/cmssw/blob/hgcal-condformat-HGCalNANO-13_2_0_pre3_linearity/RecoLocalCalo/HGCalRecAlgos/plugins/alpaka/HGCalRecHitProducer.cc
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/EDProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDPutToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESGetToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/Event.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/CopyToDevice.h"
+#include <iomanip>  // for std::setw
+#include <future>
+
+// includes for size, calibration, and configuration parameters
+#include "FWCore/Framework/interface/ESWatcher.h"
+#include "CondFormats/HGCalObjects/interface/HGCalConfiguration.h"
+#include "CondFormats/HGCalObjects/interface/HGCalMappingModuleIndexer.h"
+#include "CondFormats/DataRecord/interface/HGCalElectronicsMappingRcd.h"
+#include "CondFormats/DataRecord/interface/HGCalModuleConfigurationRcd.h"
+#include "CondFormats/HGCalObjects/interface/HGCalCalibrationParameterHost.h"
+#include "CondFormats/HGCalObjects/interface/alpaka/HGCalCalibrationParameterDevice.h"  // also for HGCalConfigParamDevice
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  using namespace cms::alpakatools;
+
+  class TestHGCalRecHitESProducers : public stream::EDProducer<> {
```

Comment:
Why do you call the module `...ESProducer` if it actually is an `EDProducer` ?

Reply:
Ah, I think I understand, because it is an `EDProducer` used to _test_ an `ESProducer` ?

Then the name is kind of OK.
Maybe `HGCalRecHitESProducersTest` would be less confusing - but for a test, I would not mind.

Reply:
OK